processing/src/main/java/io/druid/query/groupby/RowBasedColumnSelectorFactory.java

```diff
@@ -407,22 +415,40 @@ public void inspectRuntimeShape(RuntimeShapeInspector inspector)
     } else {
       return new ColumnValueSelector()
       {
+        @Override
+        public boolean isNull()
+        {
+          return row.get().getRaw(columnName) == null;
+        }
+
         @Override
         public double getDouble()
         {
-          return row.get().getMetric(columnName).doubleValue();
+          Number metric = row.get().getMetric(columnName);
+          if (NullHandling.sqlCompatible() && metric == null) {
+            throw new IllegalStateException("Cannot return double for Null Value");
```

Comment:
In follow-on patches, this exception would be avoided by having aggregators check "isNull()" before attempting to aggregate, is my understanding correct?

Reply:
yes.